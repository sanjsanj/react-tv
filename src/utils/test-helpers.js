export const messagesPayload = [
  {
    id: 'cd445e6d-e514-424f-ba8f-16ec842002c6',
    userId: 'fe27b760-a915-475c-80bb-7cdf14cc6ef3',
    message: 'Quisque erat eros, viverra eget, congue eget, semper rutrum, nulla.',
    timestamp: '2017-02-09T04:27:38Z',
  },
  {
    id: 'b03569ae-ccbf-4975-8040-4daba638b407',
    userId: '16373df5-da0a-4074-8295-f916b94269f4',
    message: 'Integer a nibh. In quis justo. Maecenas rhoncus aliquam lacus.',
    timestamp: '2016-11-09T05:04:58Z',
  },
];

export const membersPayload = [
  {
    id: 'fe27b760-a915-475c-80bb-7cdf14cc6ef3',
    firstName: 'Martin',
    lastName: 'Bradley',
    email: 'mbradley0@google.it',
    avatar: 'http://dummyimage.com/100x100.png/5fa2dd/ffffff',
    ip: '166.124.172.160',
  },
  {
    id: '16373df5-da0a-4074-8295-f916b94269f4',
    firstName: 'Helen',
    lastName: 'Hawkins',
    email: 'hhawkins1@posterous.com',
    avatar: 'http://dummyimage.com/100x100.jpg/dddddd/000000',
    ip: '179.239.189.173',
  },
];
