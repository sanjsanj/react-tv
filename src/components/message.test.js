import React from 'react';
import { shallow } from 'enzyme';

import { messagesPayload, membersPayload } from '../utils/test-helpers';

import Message from './Message';

describe('Message', () => {
  let tree;

  beforeAll(() => {
    const props = {
      data: messagesPayload[0],
      user: membersPayload[0],
    };

    tree = shallow(<Message {...props} />);
  });

  it('should match snapshot', () => {
    expect(tree).toMatchSnapshot();
  });

  it('should display message text', () => {
    const actualText = tree.find('.message').text();

    const expectedText = messagesPayload[0].message;

    expect(expectedText).toEqual(actualText);
  });

  it('should display user email when mousing over message', () => {
    const message = tree.find('.message');
    message.simulate('mouseenter');
    const userEmail = tree.find('.email');

    expect(userEmail.prop('style')).toHaveProperty('visibility', 'visible');
  });

  it('should not display user email when mouse leaves message', () => {
    const message = tree.find('.message');
    message.simulate('mouseleave');
    const userEmail = tree.find('.email');

    expect(userEmail.prop('style')).toHaveProperty('visibility', 'hidden');
  });

  it('should display the user avatar next to the message', () => {
    const userAvatar = tree.find('.avatar');

    expect(userAvatar.exists()).toEqual(true);
  });

  it('should display the message timestamp', () => {
    const timestamp = tree.find('.timestamp');
    const expectedTimestamp = new Date(messagesPayload[0].timestamp).toGMTString();

    expect(timestamp.text()).toEqual(expectedTimestamp);
  });
});
