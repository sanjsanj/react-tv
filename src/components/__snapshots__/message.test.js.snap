// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message should match snapshot 1`] = `
<li>
  <img
    alt="Martin Bradley's avatar"
    className="avatar"
    src="http://dummyimage.com/100x100.png/5fa2dd/ffffff"
  />
  <div
    className="message"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    Quisque erat eros, viverra eget, congue eget, semper rutrum, nulla.
  </div>
  <div
    className="timestamp"
  >
    Thu, 09 Feb 2017 04:27:38 GMT
  </div>
  <div
    className="email"
    style={
      Object {
        "visibility": "hidden",
      }
    }
  >
    mbradley0@google.it
  </div>
</li>
`;
